filters/colorinterp/ColorinterpFilter.cpp

```diff
@@ -0,0 +1,216 @@
+        location = name; \
+        size = sizeof(name); \
+        rampFilename = "/vsimem/" + std::string(#name) + ".png"; \
+        vsifile = VSIFileFromMemBuffer(rampFilename.c_str(), location, size, FALSE); \
+    }
+//
+std::shared_ptr<pdal::gdal::Raster> openRamp(std::string& rampFilename)
+{
+    // If the user] selected a default ramp name, it will be opened by
+    // one of these macros if it matches. Otherwise, we just open with the
+    // GDALOpen'able the user gave us
+
+    // GETRAMP will set rampFilename to the /vsimem filename it
+    // used to actually open the file, and from then on it can be treated
+    // like any other GDAL datasource.
+
+    GETRAMP(awesome_green);
+    GETRAMP(black_orange);
+    GETRAMP(blue_hue);
+    GETRAMP(blue_red);
+    GETRAMP(heat_map);
+    GETRAMP(pestel_shades);
+
+    std::shared_ptr<pdal::gdal::Raster> output (new pdal::gdal::Raster(rampFilename.c_str()));
+    return output;
+}
+
+void ColorinterpFilter::addArgs(ProgramArgs& args)
+{
+    args.add("dimension", "Dimension to interpolate", m_interpDimString, "Z");
+    args.add("minimum", "Minimum value to use for scaling", m_min);
+    args.add("maximum", "Maximum value to use for scaling", m_max);
+    args.add("ramp", "GDAL-readable color ramp image to use", m_colorramp, "pestel_shades");
+    args.add("invert", "Invert the ramp direction", m_invertRamp, false);
+    args.add("stddev", "Compute minimum/maximum given this many standard deviations", m_stdDevThreshold, 0.0);
+}
+
+void ColorinterpFilter::addDimensions(PointLayoutPtr layout)
+{
+    layout->registerDims({Dimension::Id::Red, Dimension::Id::Green, Dimension::Id::Blue});
+}
+
+void ColorinterpFilter::initialize()
+{
+    gdal::registerDrivers();
+
+    m_raster = openRamp(m_colorramp);
+    m_raster->open();
+
+    log()->get(LogLevel::Debug) << getName() << " raster connection: "
+                                             << m_raster->m_filename << std::endl;
+
+    m_interpDim = Dimension::id(m_interpDimString);
+    if (m_interpDim == Dimension::Id::Unknown)
+        throw pdal_error("Dimension name is not known!");
+}
+
+void ColorinterpFilter::filter(PointView& view)
+{
+    // If the user set a stddev value, we use that
+    // over anything else
+    if ( m_stdDevThreshold != 0.0)
+    {
+        std::vector<double> values(view.size());
+
+        pdal::stats::Summary summary(pdal::Dimension::name(m_interpDim), pdal::stats::Summary::NoEnum);
+        for (PointId idx = 0; idx < view.size(); ++idx)
+        {
+            double v = view.getFieldAs<double>(m_interpDim, idx);
+            summary.insert(v);
+            values[idx] = v;
+        }
+
+        auto median = [](std::vector<double> vals)
+        {
+            std::nth_element(vals.begin(), vals.begin()+vals.size()/2, vals.end());
+
+            return *(vals.begin()+vals.size()/2);
+        };
+
+        double med = median(values);
+        double threshold = (m_stdDevThreshold * summary.stddev());
```

Comment:
@chambbj: That's confusing.  Have no idea what MAD is, but the magic number on line 101 is evil.  Seems like the whole thing needs some comment if what you're doing is computing some bastardized stdev.  But I guess this is the wrong review for this :)